packages/slice-machine/src/legacy/lib/builders/common/Zone/index.jsx

```diff
@@ -87,63 +88,76 @@ const Zone = ({
-              (CUSTOM_NAME === undefined || f.variant === CUSTOM_NAME),
-          );
-        })
-      }
-      triggerDataTestId={
-        Boolean(isRepeatable) ? "add-field-in-items" : undefined
-      }
-    />
-  );
 
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-call
+  const addFieldDropdownFields = widgetsArrayWithCondUid
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+    .filter(Boolean)
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+    .map((widget) => {
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+      const { TYPE_NAME, CUSTOM_NAME } = widget;
+      return allFields.find(
+        (f) =>
+          f.type === TYPE_NAME &&
+          (CUSTOM_NAME === undefined || f.variant === CUSTOM_NAME),
+      );
+    });
+
+  const addFieldDropdownProps = {
+    disabled: newFieldData !== null,
+    onSelectField: onSelectFieldType,
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+    fields: addFieldDropdownFields,
+    triggerDataTestId: Boolean(isRepeatable) ? "add-field-in-items" : undefined,
+  };
   return (
     <>
       <ListHeader
         actions={
           // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
-          fields.length > 0 ? (
-            <>
-              <Text color="grey11" component="span" noWrap>
-                Show code snippets?
-              </Text>
-              <Switch
-                checked={showHints}
-                onCheckedChange={setShowHints}
-                size="small"
-                // TODO(DT-1710): add the missing `flexShrink: 0` property to the Editor's Switch component.
-                style={{ flexShrink: 0 }}
-                data-testid="code-snippets-switch"
-              />
-              {addFieldDropdown}
-            </>
-          ) : undefined
+          <>
+            {
+              // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+              fields.length > 0 && (
+                <>
+                  <Text color="grey11" component="span" noWrap>
+                    Show code snippets?
+                  </Text>
+                  <Switch
+                    checked={showHints}
+                    onCheckedChange={setShowHints}
+                    size="small"
+                    // TODO(DT-1710): add the missing `flexShrink: 0` property to the Editor's Switch component.
+                    style={{ flexShrink: 0 }}
+                    data-testid="code-snippets-switch"
+                  />
+                </>
+              )
+            }
+            {
+              // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+              zoneType === "slice" && fields.length > 0 ? (
+                <AddFieldDropdown {...addFieldDropdownProps} />
+              ) : zoneType === "customType" ? (
```

Comment:
⚠️ #issue

For me there is an issue about when I'm seeing the new static zone info dialog, for custom types.

Even the message talk about page types, did product ask us to do both custom types AND page types?

Reply:
Confirmed this with product, made the change to only display the modal for page types.